frontend/model/contracts/group.js

```diff
@@ -1360,13 +1360,81 @@ sbp('chelonia/defineContract', {
     },
     'gi.contracts/group/updateDistributionDate': {
       validate: actionRequireActiveMember(optional),
-      process ({ meta }, { state, getters }) {
-        const period = getters.periodStampGivenDate(meta.createdDate)
+      process ({ meta, contractID }, { state, getters }) {
+        const periodTo = getters.periodStampGivenDate(meta.createdDate)
+        const periodLength = getters.groupSettings.distributionPeriodLength
         const current = state.settings?.distributionDate
-        if (current !== period) {
-          // right before updating to the new distribution period, make sure to update various payment-related group streaks.
-          updateGroupStreaks({ state, getters })
-          state.settings.distributionDate = period
+        const currentPeriodPayments = cloneDeep(getters.groupPeriodPayments[current])
+
+        if (comparePeriodStamps(periodTo, current) > 0) {
```

Comment:
Before, if `current` was `undefined`, `updateGroupStreaks` would get called. Now it will no longer get called.

Reply:
Makes sense.